backend/endpoints/rooms.py

```diff
@@ -76,6 +77,88 @@ async def create_room(
     )
 
 
+@router.patch(
+    "/org/{org_id}/rooms/{room_id}/members/{member_id}",
+    response_model=ResponseModel,
+    status_code=status.HTTP_200_OK,
+    responses={
+        404: {"detail": "room or member not found"},
+        424: {"detail": "member removal unsuccessful"},
+    },
+)
+async def remove_member(
+    org_id: str, room_id: str, member_id: str, admin_id: Optional[str] = None
+):
+    """Removes a member from a room either when removed by an admin or member leaves the room.
+
+    Fetches the room which the member is removed from from the database collection
+    Pops the member being removed from the room's members dict
+    Updates the database collection with the new room
+    Returns the room dict if member was removed successfully
+
+    Args:
+        org_id (str): A unique identifier of an organisation
+        member_id (str): A unique identifier of the member being removed from the room
+        room_id (str): A unique identifier of the room a member is being removed from
+        admin_id (str): A unique identifier of the member removing another member
+
+    Returns:
+        HTTP_200_OK (member removed from room): {room}
+    Raises
+        HTTP_404_NOT_FOUND: room or member not found
+        HTTP_403_FORBIDDEN: not authorized to remove room  member
+        HTTP_424_FAILED_DEPENDENCY: member removal unsuccessful
+    """
+    room_obj = await get_room(org_id, room_id)
+    if room_obj == {}:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="room does not exist",
+        )
+    if room_obj["room_type"] != RoomType.CHANNEL:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="cannot remove member from DM rooms",
+        )
+
+    if member_id not in room_obj["room_members"]:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="user not a member of the room",
+        )
+
+    member = room_obj["room_members"].get(
+        admin_id
+    )  # member will be none if no admin is supplied
+
+    if member is not None and member.get("role") != Role.ADMIN:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="must be an admin to remove member",
+        )
+
+    try:
+        result = await remove_room_member(
+            org_id=org_id, room_data=room_obj, member_id=member_id
+        )
+        return JSONResponse(
```

Comment:
The try block should only handle the part likely to raise an exception.

The code that will be executed if no exception occurs can be in the `else` block of the `try-except-else `clause or it can just be placed outside the try-except block.

Reply:
done thanks